contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -0,0 +1,519 @@
+                )
+                id = handlers.keys[0]
+            }
+            return self.schedule(handlerCap: handlers[id!]!.capability, data: data, timestamp: timestamp, priority: priority, executionEffort: executionEffort, fees: <-fees)
+        }
+
+        /// Schedule a transaction and store it in the manager's dictionary
+        /// @param handlerCap: A capability to a resource that implements the TransactionHandler interface
+        /// @param data: Optional data to pass to the transaction when executed
+        /// @param timestamp: The timestamp when the transaction should be executed
+        /// @param priority: The priority of the transaction (High, Medium, or Low)
+        /// @param executionEffort: The execution effort for the transaction
+        /// @param fees: A FlowToken vault containing sufficient fees
+        /// @return: The ID of the scheduled transaction
+        access(Owner) fun schedule(
+            handlerCap: Capability<auth(FlowTransactionScheduler.Execute) &{FlowTransactionScheduler.TransactionHandler}>,
+            data: AnyStruct?,
+            timestamp: UFix64,
+            priority: FlowTransactionScheduler.Priority,
+            executionEffort: UInt64,
+            fees: @FlowToken.Vault
+        ): UInt64 {
+            // Clean up any stale transactions before scheduling a new one
+            self.cleanup()
+
+            // Route to the main FlowTransactionScheduler
+            let scheduledTransaction <- FlowTransactionScheduler.schedule(
+                handlerCap: handlerCap,
+                data: data,
+                timestamp: timestamp,
+                priority: priority,
+                executionEffort: executionEffort,
+                fees: <-fees
+            )
+
+            // Store the handler capability in our dictionary for later retrieval
+            let id = scheduledTransaction.id
+            let handlerRef = handlerCap.borrow()
+                ?? panic("Invalid transaction handler: Could not borrow a reference to the transaction handler")
+            let handlerTypeIdentifier = handlerRef.getType().identifier
+            let handlerUUID = handlerRef.uuid
+
+            self.handlerUUIDsByTransactionID[id] = handlerUUID
+
+            // Store the handler capability in the handlers dictionary for later retrieval
+            if let handlers = self.handlerInfos[handlerTypeIdentifier] {
+                if let handlerInfo = handlers[handlerUUID] {
+                    handlerInfo.addTransactionID(id: id)
+                    handlers[handlerUUID] = handlerInfo
+                } else {
+                    let handlerInfo = HandlerInfo(typeIdentifier: handlerTypeIdentifier, capability: handlerCap)
+                    handlerInfo.addTransactionID(id: id)
+                    handlers[handlerUUID] = HandlerInfo(typeIdentifier: handlerTypeIdentifier, capability: handlerCap)
```

Comment:
Shouldn't save the updated handlerInfo at uuid?

```suggestion
                    handlers[handlerUUID] = handlerInfo
```

Reply:
yes, good catch. Hadn't gotten to that test case yet

Reply:
added